trackintel/preprocessing/positionfixes.py

```diff
@@ -179,17 +125,20 @@ def generate_staypoints(positionfixes, method='sliding',
                 staypoint['id'] = staypoint_id
 
                 # point geometry of staypoint
-                staypoint.geometry = Point(group[name_geocol].x.mean(),
+                staypoint[name_geocol] = Point(group[name_geocol].x.mean(),
                                            group[name_geocol].y.mean())
 
                 ret_spts = ret_spts.append(staypoint, ignore_index=True)
-
-    ret_spts = gpd.GeoDataFrame(ret_spts, geometry='geom',crs=ret_pfs.crs)
+        ret_spts.set_index('id', inplace=True)
+        
+    ret_pfs = gpd.GeoDataFrame(ret_pfs, geometry=name_geocol,crs=ret_pfs.crs)
+    ret_spts = gpd.GeoDataFrame(ret_spts, geometry=name_geocol,crs=ret_pfs.crs)
     
     ## ensure dtype consistency 
     # stps id (generated by this function) should be int64
-    ret_spts['id'] = ret_spts['id'].astype('int64')
-    ret_pfs['staypoint_id'] = ret_pfs['staypoint_id'].astype('int64')
+    ret_spts.index = ret_spts.index.astype('int64')
+    # ret_pfs['staypoint_id'] should be float
+    ret_pfs['staypoint_id'] = ret_pfs['staypoint_id'].astype('float')
```

Comment:
Hm... it is a bit weird to have these different types for the different indices.... Maybe we can either set everything to float or set everything to int and use "None" for missing values...

Reply:
Anyways this is no problem that seems important at the moment...

Reply:
I think we cannot use int for mixed int and np.nan columns, [here](https://pandas.pydata.org/pandas-docs/stable/user_guide/gotchas.html#support-for-integer-na). Simple workaround is to use float. I should add more comments here